tests/src/python/test_qgsserver_wms.py

```diff
@@ -183,6 +183,40 @@ def test_getcapabilities_advertised_url(self):
             b"\n".join(headers) + b"\n\n" + bytes(response.body()), expected
         )
 
+    def test_getcapabilities_scale_denominator(self):
+        server = QgsServer()
+        request = QgsServerRequest()
+        projectPath = os.path.join(
+            self.testdata_path, "test_project_scale_denominator.qgs"
+        )
+        request.setUrl(
+            QUrl(
+                "http://localhost/qgis_mapserv.fcgi?MAP="
+                + projectPath
+                + "&SERVICE=WMS&REQUEST=GetCapabilities"
+            )
+        )
+        request.setOriginalUrl(QUrl("http://localhost/wms/test_project"))
+        response = QgsBufferServerResponse()
+        server.handleRequest(request, response)
+        response.flush()
+
+        headers = []
+        rh = response.headers()
+        rk = sorted(rh.keys())
+        for k in rk:
+            headers.append((f"{k}: {rh[k]}").encode())
+
+        reference_path = os.path.join(
+            self.testdata_path, "wms_getcapabilities_scale_denominator.txt"
+        )
+        f = open(reference_path, "rb")
+        expected = f.read()
+        print(bytes(response.body()))
+        self.assertXMLEqual(
```

Comment:
can you use `wms_request_compare` as all other test cases do?

Reply:
Done.